Code/Tools/FBuild/FBuildCore/Graph/Node.h

```diff
@@ -110,8 +110,11 @@ class Node : public Struct
         NOT_PROCESSED,      // no work done (either not part of this build, or waiting on static dependencies )
         PRE_DEPS_READY,     // pre-build deps processed
         STATIC_DEPS_READY,  // static dependencies are uptodate - we are ready to DoDynamicDeps
+		PRE_DYN_DEPS_READY, // pre-build dynamic deps list updated
+		PRE_DYN_DEPS_DONE,  // pre-build dynamic deps list has been updated, wait for pre build dynamic deps to be ready
         DYNAMIC_DEPS_DONE,  // dynamic deps updated, waiting for dynamic deps to be ready
         BUILDING,           // in the queue for building
+		PRE_DYN_DEPS_BUILDING, // use the job queue to build dynamic dependencies
```

Comment:
This stage actually happens between PRE_DYN_DEPS_READY and PRE_DYN_DEPS_DONE, however I put it after BUILDING so that all of the logic that checks to see if this node is currently in a job queue can stay as-is. There are multiple spots where we check if state > BUILDING.